blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/EventElasticSearchIO.java

```diff
@@ -0,0 +1,122 @@
+        this(RemoteElasticSearchServer.getInstance());
+    }
+    public EventElasticSearchIO(Client client) {
+        this.client = client;
+    }
+    public EventElasticSearchIO(ElasticClientManager manager) {
+        this(manager.getClient());
+    }
+
+    @Override
+    public void insert(String tenant, List<Map<String, Object>> events) throws Exception {
+        BulkRequestBuilder bulk = client.prepareBulk();
+
+        for (Map<String, Object> event : events) {
+            event.put(Event.FieldLabels.tenantId.toString(), tenant);
+            IndexRequestBuilder requestBuilder = client.prepareIndex(EVENT_INDEX, ES_TYPE)
+                    .setSource(event)
+                    .setRouting(tenant);
+            bulk.add(requestBuilder);
+        }
+        bulk.execute().actionGet();
+    }
+
+    @Override
+    public List<Map<String, Object>> search(String tenant, Map<String, List<String>> query) throws Exception {
+        BoolQueryBuilder qb = boolQuery()
+                .must(termQuery(Event.FieldLabels.tenantId.toString(), tenant));
+
+        if (query != null) {
+            qb = extractQueryParameters(query, qb);
+        }
+
+        SearchResponse response = client.prepareSearch(EVENT_INDEX)
+                .setRouting(tenant)
+                .setSize(100000)
+                .setVersion(true)
+                .setQuery(qb)
+                .execute()
+                .actionGet();
+
+        List<Map<String, Object>> events = new ArrayList<Map<String, Object>>();
+        for (SearchHit hit : response.getHits().getHits()) {
+            events.add(hit.getSource());
+        }
+
+        return events;
+    }
+
+    private BoolQueryBuilder extractQueryParameters(Map<String, List<String>> query, BoolQueryBuilder qb) {
+        String tagsQuery = extractFieldFromQuery(Event.FieldLabels.tags.toString(), query);
+        String untilQuery = extractFieldFromQuery(Event.untilParameterName, query);
+        String fromQuery = extractFieldFromQuery(Event.fromParameterName, query);
+
+        if (!tagsQuery.equals(""))
+            qb = qb.must(termQuery(Event.FieldLabels.tags.toString(), tagsQuery));
+
+        if (!untilQuery.equals("") && !fromQuery.equals("")) {
+            qb = qb.must(rangeQuery(Event.FieldLabels.when.toString())
+                    .to(Long.parseLong(untilQuery))
+                    .from(Long.parseLong(fromQuery)));
+        } else if (!untilQuery.equals("")) {
+            qb = qb.must(rangeQuery(Event.FieldLabels.when.toString()).to(Long.parseLong(untilQuery)));
+        } else if (!fromQuery.equals("")) {
+            qb = qb.must(rangeQuery(Event.FieldLabels.when.toString()).from(Long.parseLong(fromQuery)));
+        }
+
+        return qb;
+    }
+
+    private String extractFieldFromQuery(String name, Map<String, List<String>> query) {
+        String result = "";
+        if (query.containsKey(name)) {
+            try {
+                result = query.get(name).get(0);
```

Comment:
It does not work currently, we tried it for tag. The query would return a 200 with no result.